Character.cpp

```diff
@@ -60,43 +56,21 @@ Character* Character::parseUnit(const std::string& FileName)
 	else {
 		throw 1;
 	}
-	return new Character(name, hp, dmg);
+	return Character(name, hp, dmg);
 }
 
 bool Character::IsDead() const
 {
 	return hp == 0;
 }
 
-void Character::DMGTaken(Character*& foe)
+void Character::DMGTaken(Character* foe)
```

Comment:
Itt még pointert vár paraméterként.

Reply:
@oliverosz Hogyan tudnám akkor a "this" mutatót helyesen referenciaként átadni, ebben kérhetnék segítséget?

Reply:
A this mutatót sehogy. Az általa mutatott objektumot add át referenciaként.

Reply:
@oliverosz Kérlek még ebben segítenél hogyan tudnám a this által mutatott objektumot átadni referenciaként, mert hiába próbálom azóta nem jövök rá hogyan kellene? Köszönöm a válaszod előre is!

Reply:
Remélem jól sikerült javítanom ezt a hibát végül.